packages/sync/src/modules/class-comments.php

```diff
@@ -181,12 +197,54 @@ public function get_whitelisted_comment_types() {
 	/**
 	 * Prevents any comment types that are not in the whitelist from being enqueued and sent to WordPress.com.
 	 *
-	 * @param array $args Arguments passed to wp_insert_comment.
+	 * @param array $args Arguments passed to wp_insert_comment, deleted_comment, spammed_comment, etc.
 	 *
-	 * @return bool or array $args Arguments passed to wp_insert_comment
+	 * @return bool or array $args Arguments passed to wp_insert_comment, deleted_comment, spammed_comment, etc.
 	 */
 	public function only_allow_white_listed_comment_types( $args ) {
-		$comment = $args[1];
+		$comment = false;
+
+		if ( isset( $args[1] ) ) {
+			// comment object is available.
+			$comment = $args[1];
+		} else {
+			// comment_id is available.
+			$comment = get_comment( $args[0] );
+		}
```

Comment:
Do you think it would be better to write this condition as: 

```
if ( is_numeric( $args[0] ) ) {
    $comment = get_comment( $args[0] );
}
elseif ( isset( $args[1] ) ) {
    $comment = $args[1];
}
```

Reply:
we want to do the args[1] check before args[0] as it will save us a get_comment call since the object will already be available. validation of args[0] would be done in the get_comment call but no harm in adding an is_numeric check on it.

Will update.